include/neml2/misc/utils.h

```diff
@@ -25,11 +25,14 @@
 #pragma once
 
 #include "neml2/misc/types.h"
+#include "neml2/misc/error.h"
 
 namespace neml2
 {
 namespace utils
 {
+std::string demangle(const char * name);
```

Comment:
I guess having this all here is okay, but do we use these new functions for anything other than parsing?  If not then maybe we should just stick them all in something like `parser_utils` or similar.

Reply:
I moved all of them to `parser_utils`